.circleci/config.yml

```diff
@@ -17,7 +17,85 @@ workflows:
                 - master
             tags:
               only: /^\d+\.\d+\.\d+$/
+      - publish_python:
+          requires:
+            - compile
+          filters:
+            branches:
+              only:
+                - master
+            tags:
+              only: /^\d+\.\d+\.\d+$/
 jobs:
+  publish_python:
+    docker:
+      - image: python:3.6-alpine
+    environment:
+      PYTHON_PROTO_PATH: /tmp/python_protos
+    steps:
+      - checkout
+      - run:
+          name: Install git
+          command: |
+            apk add git
+      - run:
+          name: Install protoc
+          command: |
+            apk add protoc
+      - run:
+          name: Install OpenSSL
+          command: |
+            apk add openssl
+      - run:
+          name: Compile protos
+          # IAM is published separately because it is Cloud specific.
+          command: |
+            mkdir -p $PYTHON_PROTO_PATH
+            protoc --proto_path=. --python_out=$PYTHON_PROTO_PATH $(find google -not -path "*/iam/*" -name "*.proto")
+      - run:
+          name: Install packaging and publishing tools
+          commands: |
+            apk add pip
+            pip install setuptools twine
+
+      - run:
+          name: Decrypt PyPi config file
+          command: |
+            # We don't just use an environment variable to store the key because of character constraints in CircleCI env vars.
+            # The indirect key is much longer as a result.
+            openssl aes-256-ctr -pbkdf2 -d -k "${PYPI_PUBLISH_INDIRECT_KEY}" < ./circleci/.pypirc.enc > /tmp/.pypirc
+      - run:
+          name: Package the Python bindings
+          command: |
+            set -euo pipefail
+            # Determine package verion info
+            PROTOS_VERSION=$(git describe --tags)
+            PACKAGING_FILES=$(realpath ./.circleci/packaging/python/)
+            # Decrypt the keyfile
+            # Create __init__.py files
+            cd $PYTHON_PROTO_PATH
+            cat << EOF > google/__init__.py
+            __import__('pkg_resources').declare_namespace(__name__)
+            EOF
+            for d in $(find google -not -path google -type d); do cp google/__init__.py ${d}; done
+            # Add packaging files from templates
+            cp -r $PACKAGING_FILES .
+            # Interpolate version info
+            sed -i "s/{PROTOS_VERSION}/v1.0.0/g" $(find . -type f -not -path "./google/*")
```

Comment:
`v1.0.0` is hard coded?  (I don't really understand what is going on here)

Reply:
also, I feel just a little bit uncomfortable about blindly replacing the version in all the files - is it possible that anything could go wrong here?

Reply:
Woah, good catch.
The templates all have the string `{PROTOS_VERSION}` instead of a real version, so this step interpolates that string into the tagged version (or a hardcoded version :P)

Reply:
Oh, I see, I did not notice it does not have a `$` sign :) Looks good now.

Reply:
Only the packaging files have the version interpolated: all the generated py files are not touched.